File: loss/BaseLoss.m
classdef (Abstract) BaseLoss < handle
    %BASELOSS Summary of this class goes here
    %   Detailed explanation goes here
    
    properties (Access = 'protected')        
    end
    
    methods (Abstract, Access = 'public')
       [loss, gradients] = GetLossAndGradients(obj, scores, targets); 
    end
    
end

File: utils/optimization_functions/cppPrj/loss/baseloss.cpp
#include "baseloss.h"


tuple<float, Tensor<float> > BaseLoss::GetLossAndGradients(const Tensor<float> &scores, const Tensor<float> &targets)
{
    cout << "From BaseLoss" << endl;
    return make_tuple(0.1F,Tensor<float>());
}
